src/MSBuild/CommandLineSwitches.cs

```diff
@@ -394,6 +396,8 @@ private struct DetectedParameterizedSwitch
         private DetectedParameterizedSwitch[] _parameterizedSwitches;
         // NOTE: the above arrays are instance members because this class is not required to be a singleton
 
+        internal static Dictionary<string, string> SwitchesFromResponseFiles = new();
```

Comment:
It doesn't...would it be sufficient to just set it to null after the command line parsing step is over? I can't attach it to an individual CommandLineSwitches object because we make multiple, and I'd need to somehow know about all the other ones and have them share what they'd parsed.

Reply:
That's definitely better if you can't create it as a local and pass it to the individual objects.

Reply:
If I make it as a local, it will live for the entire build unless I move around a lot of code. The function I would make it in is Xmake.Execute.